lib/devcycle-ruby-server-sdk/event_queue.rb

```diff
@@ -0,0 +1,128 @@
+require 'typhoeus'
+require 'sorbet-runtime'
+require 'concurrent-ruby'
+
+module DevCycle
+  class EventQueue
+    extend T::Sig
+
+    attr_accessor :api_client
+    attr_accessor :events_api_uri
+    attr_accessor :event_flush_interval_ms
+    attr_accessor :flush_event_queue_size
+    attr_accessor :flush_timer_task
+    attr_accessor :flush_in_progress
+    attr_accessor :local_bucketing
+    attr_accessor :logger
+    attr_accessor :max_event_queue_size
+    attr_accessor :sdk_key
+
+
+    sig { params(sdk_key: String, options: EventQueueOptions, local_bucketing: LocalBucketing).returns(NilClass) }
+    def initialize(sdk_key, options, local_bucketing)
+      @sdk_key = sdk_key
+      @events_api_uri = options.events_api_uri
+      @event_flush_interval_ms = options.event_flush_interval_ms || 10 * 1000
```

Comment:
Should we handle this defaulting in the DVC options as a getter instead?

Reply:
yeah, that sounds good to me. I'll also move all the error checking there too